git/README.md

```diff
@@ -4,6 +4,7 @@ A guide for programming within version control.
 
 ## Best Practices
 
+- Treat the version history as a resource for the future.
```

Comment:
Can you elaborate a bit more here? I love what you have in the PR description although that's probably too long. Maybe some kind of happy medium?

Reply:
I couldn't work out how to shorten it to fit in a line or two. Any ideas?

Reply:
Maybe something along the lines of "Explain the _why_ and not the what"?

Reply:
An idea for incorporating some of the content from the PR description

```suggestion
- Treat the version history as a resource for the future. Structure your commits such that they tell the story of _why_ the change was made, so that future developers--including yourself!--can understand the constraints which influenced your decisions at the time. 
```

Reply:
In #738 we introduced the pattern of adding a markdown file with the reasoning and examples. You might consider doing the same.

Those would be nice blog posts too.

Reply:
@nickcharlton what do you think about @jutonz's suggestion and then merging this?